processing/src/main/java/org/apache/druid/segment/incremental/IncrementalIndexColumnSelectorFactory.java

```diff
@@ -91,7 +92,14 @@ class IncrementalIndexColumnSelectorFactory implements ColumnSelectorFactory, Ro
       @Override
       public ColumnCapabilities getColumnCapabilities(String column)
       {
-        return capabilitiesMap.get(column);
+        final ColumnCapabilities capabilities = capabilitiesMap.get(column);
+
+        DruidException.conditionalDefensive(
+            cursorBuildSpec.getPhysicalColumns() == null || capabilities != null || capabilitiesMap.containsKey(column),
+            "Asked for physical column capabilities for column[%s] which wasn't specified as required by the query, this is a bug",
```

Comment:
Not necessary to say "this is a bug"; it's implied by the category being "defensive".